tests/core/conversations_worker/integration/test_conversation_integration.py

```
@@ -406,6 +423,86 @@ def test_max_concurrent_never_exceeded(self, supabase_fx: SupabaseFixture):
+        """
+        num = 12  # > 10, and well above the default MAX_CONCURRENT (5)
+        conv_ids = []
+        for i in range(1, num + 1):
+            conv = supabase_fx.create_conversation(
+                ask=f"What is {i * 9} + {i}? Answer with just the number.",
+                title=f"integ: burst-p2r-{i}",
+            )
+            conv_ids.append(conv["conversation_id"])
+
+        # Per-conversation ordered history of distinct statuses observed.
+        history: Dict[str, List[str]] = {cid: [] for cid in conv_ids}
```

Ruff (0.15.18)

[error] Undefined name `Dict`

(F821)